src/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -0,0 +1,50 @@
+package org.javarosa.core.model.instance;
+
+import org.javarosa.xml.ElementParser;
+import org.javarosa.xml.TreeElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class ExternalDataInstance extends DataInstance {
+    private TreeElement root;
+
+    public ExternalDataInstance(String srcLocation, String instanceId) {
+        super(instanceId);
+        setName(instanceId);
+        try {
+            URI uri = new URI(srcLocation);
+            if ("jr".equals(uri.getScheme()) && "file".equals(uri.getHost())) {
+                String absolutePath = /* ToDo: find out how to get the actual location */
+                        System.getProperty("user.dir") + "/resources" + uri.getPath();
+                KXmlParser xmlParser = ElementParser.instantiateParser(new FileInputStream(absolutePath));
+                TreeElementParser treeElementParser = new TreeElementParser(xmlParser, 0, instanceId);
+                root = treeElementParser.parse();
```

Comment:
@dcbriccetti I think it's a bad practice in a lot of languages. The biggest problem is that it may be hard to unit test this class in isolation. It is implicitly dependent on `TreeElementParser` so you cannot use a stubbed one. It also hardcodes the path so you cannot easily change it for testing or when using JavaRosa in a client app. 

I think it also breaks the single responsibility principle where a class that stands for a tree structure is also responsible for I/O, xml parsing and handling errors that come from this operations.

Another thing: I/O calls may be considered to have side-effects in similar way network calls does (communication with outside world and errors) although they are far less likely. However, consider running this code on an Android device when user ejects an sd-card. Or on an invalid external instance xml.